### PR TITLE
♻️ Frontend env free

### PR DESCRIPTION
**Pull request by AntoLC**
## Purpose

Until now, the front had to know at build time the url of the backend to be able to communicate with it.
It is not optimal because it means that we need multiple docker image (1 per environment) to have the app working, it is not very flexible.
This PR will make the frontend "environment free" by determining these urls at runtime.

## Proposal

- [x] ♻️(frontend) frontend environment free
- [x] 🐛(frontend) fix duplicate call on getMe

### CODE REVIEW

**Review by lebaudantoine — COMMENTED** [review]
minor comments !! great worksssss

File: src/frontend/apps/desk/src/api/conf.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const baseApiUrl = (apiVersion: string = '/api/v1.0/') => {
+  const origin =
+    process.env.NEXT_PUBLIC_API_ORIGIN ||
+    (typeof window !== 'undefined' ? window.location.origin : '');
+
+  return `${origin}${apiVersion}`;
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/api/conf.ts]: https://github.com/numerique-gouv/people/pull/233/commits/d772409a3765ff3a50acf1818555c036a1538464

File: src/frontend/apps/desk/src/api/fetchApi.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import { useAuthStore } from '@/core/auth';
+import { useAuthStore } from '@/core';
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/api/fetchApi.ts]: https://github.com/numerique-gouv/people/commit/d772409a3765ff3a50acf1818555c036a1538464

File: .github/workflows/docker-hub.yml [hunk 3]
```diff
@@ -5,6 +5,7 @@ on:
   push:
     branches:
       - 'main'
+      - 'refacto/frontend-env-free'
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply .github/workflows/docker-hub.yml]: It is for development testing, I don't have Tilt yet, I need a way to test if my code works in real situation.

[the remark this answers is not in the document]
Comment (lebaudantoine) [reply .github/workflows/docker-hub.yml]: I completely understand your point, but we should probably not merge them right ?

[the remark this answers is not in the document]
Comment (AntoLC) [reply .github/workflows/docker-hub.yml]: I guess you need to rebase in local, it is not in the branch anymore.

[the remark this answers is not in the document]
Comment (lebaudantoine) [reply .github/workflows/docker-hub.yml]: I misunderstood your comment! good

**Review by lebaudantoine — APPROVED** [review]

**Pull request merged**